src/mbed_tools/build/flash.py

```diff
@@ -0,0 +1,58 @@
+#
+# Copyright (C) 2020 Arm Mbed. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+"""Flash binary onto the connected device."""
+
+import shutil
+import os
+import pathlib
+from typing import Optional
+
+from mbed_tools.devices.devices import get_connected_devices
+from mbed_tools.build.exceptions import BinaryFileNotFoundError, DeviceNotFoundError
+
+
+def flash_dev(
+    disk: pathlib.Path, image_path: pathlib.Path, copy_method: Optional[str] = "default", port: Optional[str] = None, program_cycle_s: Optional[int] = 4
+) -> None:
+    """Flash device using copy method.
+
+    Args:
+        disk: Device mount point.
+        image_path: Image file to be copied to device.
+        copy_method: Switch -c <copy_method> (default: shell).
+        port: Port.
+        program_cycle_s: Wait after flashing delay (sec).
+    """
+    try:
+        shutil.copy(image_path, disk, follow_symlinks=False)
+        os.sync()
+    except OSError:
+        raise
```

Comment:
Can we just remove the `try/except` block completely? Do we gain anything by catching this error if we're just going to raise it again?

Reply:
Removed.